EastBarley/Controllers/InvoicesController.cs

```diff
@@ -174,5 +176,24 @@ public IActionResult CreateNewOrder(int UserId, LineItems lineItemToAdd)
             }
             return Created("", cart);
         }
+
+        [HttpPut("cart/purchase")]
+        public IActionResult CompleteOrder(Invoices invoiceToComplete)
+        {
+            var completedInvoice = _repository.CompleteOrder(invoiceToComplete);
+            var deleteQuantity = _repository.GetQuantityToDelete(invoiceToComplete.InvoiceId);
+            foreach (var item in deleteQuantity)
+            {
+                item.Quantity = item.Quantity * -1;
+                _productsRepository.UpdateProductQuantity(item);
+            }
+            if (completedInvoice == null)
```

Comment:
completedInvoice will never be null because it is a list, this needs to be !completedInvoice.Any()
Everything else looks good!